src/rpc.cpp

```diff
@@ -173,20 +215,71 @@ RPCServer::~RPCServer() {
         JSValueUnprotect(m_context, item.second);
     }
 
+    get_rpc_server(m_context) = nullptr;
     JSGlobalContextRelease(m_context);
 }
 
-json RPCServer::perform_request(std::string name, json &args) {
-    try {
-        RPCRequest action = m_requests[name];
-        assert(action);
+void RPCServer::run_callback(JSContextRef ctx, JSObjectRef this_object, size_t argc, const JSValueRef arguments[], jsc::ReturnValue &return_value) {
+    RPCServer* server = get_rpc_server(JSContextGetGlobalContext(ctx));
+    if (!server) {
+        return;
+    }
+
+    // The first argument was curried to be the callback id.
+    RPCObjectID callback_id = jsc::Value::to_number(ctx, arguments[0]);
+    JSObjectRef arguments_array = jsc::Object::create_array(ctx, argc - 1, argc == 1 ? nullptr : arguments + 1);
+    json arguments_json = server->serialize_json_value(arguments_array);
+
+    // The next task on the stack will instruct the JS to run this callback.
+    server->m_worker.add_task([&]() -> json {
```

Comment:
It is executed before leaving this function. I will add a comment that confirms that.